xchange-coingi/src/main/java/org/knowm/xchange/coingi/service/CoingiMarketDataService.java

```diff
@@ -0,0 +1,63 @@
+package org.knowm.xchange.coingi.service;
+
+import org.knowm.xchange.Exchange;
+import org.knowm.xchange.coingi.CoingiAdapters;
+import org.knowm.xchange.coingi.dto.marketdata.CoingiOrderBook;
+import org.knowm.xchange.currency.CurrencyPair;
+import org.knowm.xchange.dto.marketdata.OrderBook;
+import org.knowm.xchange.dto.marketdata.Ticker;
+import org.knowm.xchange.dto.marketdata.Trades;
+import org.knowm.xchange.service.marketdata.MarketDataService;
+
+import java.io.IOException;
+
+public class CoingiMarketDataService extends CoingiMarketDataServiceRaw
+    implements MarketDataService {
+  public CoingiMarketDataService(Exchange exchange) {
+    super(exchange);
+  }
+
+  @Override
+  public Ticker getTicker(CurrencyPair currencyPair, Object... objects) throws IOException {
+    OrderBook orderBook = getOrderBook(currencyPair, 1, 1, 1);
+
+    return new Ticker.Builder()
+        .currencyPair(orderBook.getAsks().get(0).getCurrencyPair())
+        .ask(orderBook.getAsks().get(0).getLimitPrice())
+        .bid(orderBook.getBids().get(0).getLimitPrice())
+        .build();
```

Comment:
I don't think this is a ticker - the most important fields are missing
its better to throw an exception here then to return this

Reply:
(If we throw an exception here, what about the unit test / integration test that currently depend on this ticker)? 
Yes, I agree this is not a true ticker. It functions pretty closely, though. We get the best ASK and BID with every call. They are always latest. It's better to know the best ask / bid than nothing. Do you really want an exception (i.e. nothing about best ask/best bid)? It only doesn't fill statistics about avg. low, high. It gives us last (best) ask / bid. I'll let you decide. Please confirm the radical decision.

Reply:
the most important data in a ticker is OHLC data - Open High Low Close - https://en.wikipedia.org/wiki/Open-high-low-close_chart
they are used in candle stick charts - last top bid and ask are just an addition to this object not its core
you should write unit and integrations tests for code that is more meaningful as this method should `throw new NotAvailableFromExchangeException();`

Reply:
done, thanks

Reply:
done